src/mumble/UserLocalNicknameDialog.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#include "UserLocalNicknameDialog.h"
+#include "ClientUser.h"
+#include "Database.h"
+#include "MainWindow.h"
+
+#include <QtGui/QCloseEvent>
+#include <QtWidgets/QPushButton>
+
+// We define a global macro called 'g'. This can lead to issues when included code uses 'g' as a type or parameter name
+// (like protobuf 3.7 does). As such, for now, we have to make this our last include.
+#include "Global.h"
+
+UserLocalNicknameDialog::UserLocalNicknameDialog(unsigned int sessionId,
+											 QMap< unsigned int, UserLocalNicknameDialog * > *qmUserNicknameTracker)
+	: QDialog(nullptr), m_clientSession(sessionId), m_qmUserNicknameTracker(qmUserNicknameTracker) {
+	setupUi(this);
+	qleUserLocalNickname->setAccessibleName(tr("User nickname"));
+
+	ClientUser *user = ClientUser::get(sessionId);
+	if (user) {
+		QString title = tr("Adjusting local nickname for %1").arg(user->qsName);
+		setWindowTitle(title);
+		qleUserLocalNickname->setText(user->getLocalNickname());
+		m_originalNickname = qleUserLocalNickname->text();
+	}
+
+	if (g.mw && g.mw->windowFlags() & Qt::WindowStaysOnTopHint) {
+		// If the main window is set to always be on top of other windows, we should make the
+		// nickname dialog behave the same in order for it to not get hidden behind the main window.
+		setWindowFlags(Qt::WindowStaysOnTopHint);
+	}
+}
+
+void UserLocalNicknameDialog::closeEvent(QCloseEvent *event) {
+	m_qmUserNicknameTracker->remove(m_clientSession);
+	event->accept();
+}
+
+void UserLocalNicknameDialog::present(unsigned int sessionId,
+									QMap< unsigned int, UserLocalNicknameDialog * > *qmUserNicknameTracker) {
+	if (qmUserNicknameTracker->contains(sessionId)) {
+		qmUserNicknameTracker->value(sessionId)->raise();
+	} else {
+		UserLocalNicknameDialog *userNickname = new UserLocalNicknameDialog(sessionId, qmUserNicknameTracker);

```

Comment:
Instead of using a raw pointer here, it'd probably be better to use a smart pointer instead. Presumably a `std::unique_ptr` would be what we want here:
```suggestion
		std::unique_ptr< UserLocalNicknameDialog > userNickname = std::make_unique< UserLocalNicknameDialog >(sessionId, qmUserNicknameTracker);
```
You probably have to include the `<memory>` header for this to compile :point_up: 

Note also that you'd then have to `move` the pointer into the map (whose type has to be adjusted as well).